### DISCUSSION TITLE
Error handling/logging with MAUI

### DISCUSSION THREAD

**Discussion by ThomasArdal** [dotnet/maui#1030] [Q&A]
I couldn't find any documentation on global error handling/logging with MAUI. With Xamarin and Xamarin.Forms you would do something like this (Android as an example):

```csharp
AndroidEnvironment.UnhandledExceptionRaiser += (sender, e) =>
{
    // do something with e
    e.Handled = true;
};
TaskScheduler.UnobservedTaskException += (sender, e) =>
{
    // do something with e
    e.SetObserved();
};
```

It seems to work in MAUI too. But I'm wondering if there are better ways to create these hooks now. Maybe as part of the `appBuilder` in `Startup.cs`?

**Comment by ThomasArdal** [comment]
Something like this would be nice:

```csharp
appBuilder
    .UseFormsCompatibility()
    .UseMauiApp<App>()
    .ConfigureLogging((ctx, logging) =>
    {
        logger.Add* <-----
    })
    .ConfigureFonts(...);
```

**Comment by marwalsch** [comment]
A platform agnostic option similar to `IApplicationBuilder.UseDeveloperExceptionPage()` would also be nice to have, preferably one that accepts a delegate parameter to customize global handling behaviour.
